### PR TITLE
E2etest

### PR DESCRIPTION
**Pull request by facoinbase**
### What changed? Why?
Add examples for solana stake, unskake and claim operations

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: src/examples/solana_stake.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  } else if (networkID == "solana-devnet") {
+    return "https://explorer.solana.com/tx/" + signature + "?cluster=devnet";
+  }
+
+  return ""
+}
+
+Coinbase.configureFromJson({ filePath: coinbaseApiKeyPath() });
+
+const kp = getKeypair();
```
Comment (ProfMoo) [inline src/examples/solana_stake.ts:93]: Can we use more descriptive/complete variable names in these examples across the board? AFAIK, it's more common to be descriptive in variable names in TS than Golang - we want to follow the SDK language's industry standards.

File: src/examples/solana_stake.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import { StakingReward } from "../coinbase/staking_reward";
+import { Coinbase } from "../coinbase/coinbase";
+import { Address } from "../coinbase/address";
+import { NetworkIdentifier, StakingOperation } from "../client";
+import { coinbaseApiKeyPath, getKeypair } from "./solana_wallet";
+import { KeyPairSigner, createSolanaRpc, decodeTransactionMessage, getBase58Codec, getBase58Decoder, getBase64EncodedWireTransaction, getBase64Encoder, getCompiledTransactionMessageCodec, getSignersFromTransactionMessage, getTransactionDecoder, sendAndConfirmTransactionFactory, signAndSendTransactionMessageWithSigners, signTransaction } from "@solana/web3.js";
+import { TextEncoder } from "util";
+import * as bs58 from 'bs58';
+
+// import { StakingOperation } from "../coinbase/staking_operation";
```
Comment (ProfMoo) [inline src/examples/solana_stake.ts:11]: Let's remove commented code.

File: src/examples/solana_stake.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        options: { amount: "100000000" },
+      }))!.data!);
+  }
+
+  console.log("Unstakeable: ");
+  console.log(ctx?.unstakeable_balance.amount);
+  console.log(ctx?.unstakeable_balance.asset);
+
+  {
+    const amount = ctx?.unstakeable_balance.amount ? BigInt(ctx?.unstakeable_balance.amount.toString()) : 0;
```
Comment (ProfMoo) [inline src/examples/solana_stake.ts:38]: Seeing a lot of optional chaining in these examples. Is this OK for examples, or should we attempt to remove these for clarity?

Comment (facoinbase) [reply src/examples/solana_stake.ts:38]: This is just for output formatting, I think it is fine

**Review by ProfMoo — APPROVED** [review]
This LGTM, but @John-peterson-coinbase might want to take a look since we're changing the `package.json`

**Comment by John-peterson-coinbase** [comment]
These look reasonable to me.

@ProfMoo @facoinbase - Can we move these to the `./quickstart-template` package and ensure they are executable. We are planning on deprecating the `src/examples` dir and using this as the main place to put executable examples as we discussed on slack. Thanks!

File: quickstart-template/package.json
Comment (John-peterson-coinbase) [inline quickstart-template/package.json:1]: please add commands like the other quick starts to run the new quick starts.

**Review by John-peterson-coinbase — COMMENTED** [review]
Can we also add some documentation comments similar to the other quick starts (`./register-basename.js` is a good example) to help developers understand what each function is doing.

File: quickstart-template/package.json
Comment (facoinbase) [reply quickstart-template/package.json:1]: done

**Pull request merged**